src/utils/useMouseTracking.js

```diff
@@ -51,21 +74,17 @@ export const useMouseTracking = () => {
 				if (mouseTrackingActive && data.pageX != undefined && data.pageY != undefined) {
 					deviceRef.current = await getUserDevice();
 
-					if (!batchReferenceRef.current) {
-						batchReferenceRef.current = doc(
-							db,
-							`users/${user.uid}/visitedPages/${data.currentWebPageTitle}/mouseTracking`,
-							'mouseData',
-						);
-					}
+					batchReferenceRef.current = doc(
+						db,
+						`users/${user.uid}/visitedPages/${data.currentWebPageTitle}/mouseTracking`,
+						'mouseData',
+					);
 
-					if (!batchSnapRef.current) {
-						batchSnapRef.current = await getDoc(batchReferenceRef.current);
-						if (!batchSnapRef.current.exists()) {
-							await setDoc(batchReferenceRef.current, {
-								values: [],
-							});
-						}
+					batchSnapRef.current = await getDoc(batchReferenceRef.current);
+					if (!batchSnapRef.current.exists()) {
+						await setDoc(batchReferenceRef.current, {
+							values: [],
+						});
 					}
```

Comment:
Als je die async function elke keer gaat uitvoeren moet je dat niet saven in een ref.